src/sync.rs

```diff
@@ -266,6 +281,247 @@ impl Interned {
     }
 }
 
+/// TODO
+pub struct RemoteAllowThreads<'py>(pub(crate) Python<'py>);
+
+impl<'py> RemoteAllowThreads<'py> {
+    /// TODO
+    ///
+    /// # Safety
+    ///
+    /// TODO
+    pub unsafe fn local(self) -> LocalAllowThreads<'py> {
+        LocalAllowThreads(self.0)
+    }
+
+    /// TODO
+    ///
+    /// # Safety
+    ///
+    /// TODO
+    pub unsafe fn local_unconstrained(self) -> LocalUnconstrainedAllowThreads<'py> {
```

Comment:
I guess that the caller now writes

```rust
unsafe { py.allow_threads().local_unconstrained() }.with(|| ...)
```

Which looks good to me 👍. I guess we would need to document this style to help users avoid accidentally infecting the closure with unsafe.

I like the consistency of calling `.with` in all cases, and the proposed names make sense to me 🎉

Reply:
Yes, this is exactly the style I ended up using in the benchmarks. I will document all thread options in the doc comment for `allow_threads` and emphasize the "infected closure" problem...